### PR TITLE
Allow multiple facebook apps (per domain)

### LINKED ISSUE #430: FB and Twitter integration doesn't work with dashboard.onzestadapp.be
We can not add multiple domains to the Rogerthat facebook app so we'll have to come up with a different solution.

Creating different apps for the dashboard.onzestadapp.be and rogerthat-server.appspot.com domains?

### PR DESCRIPTION
**Pull request by abom**
Closes #430.

Should wait until the new apps get approved after Facebook app review.

### CODE REVIEW

File: src/static/js/solutions/sln-settings.js [hunk 1]
```diff
@@ -406,6 +406,10 @@ $(function () {
             url: '/common/settings/facebook/app/id',
             method: 'get',
             success: function(app_id) {
+                if(!app_id) {
+                    console.error('Cannot get facebook app id');
```
Comment (bart-at-mobicage) [inline src/static/js/solutions/sln-settings.js]: Don't silently die. Show the "unexpected error occurred. please try again later" popup.

File: src/solutions/common/bizz/facebook/__init__.py [hunk 2]
```diff
@@ -48,11 +63,12 @@ def post_to_facebook(access_token, message, image=None):
...
-def extend_access_token(access_token):
-    server_settings = get_solution_server_settings()
-    app_id = server_settings.facebook_app_id
-    app_secret = server_settings.facebook_app_secret
+@arguments(domain=unicode, access_token=unicode)
+def extend_access_token(domain, access_token):
+    try:
+        app_id, app_secret = get_facebook_app_info(domain)
+    except ValueError:
+        raise ValueError('No facebook app is linked to %s' % domain)
```
Comment (bart-at-mobicage) [inline src/solutions/common/bizz/facebook/__init__.py]: Is it possible that a ValueError is raised?

File: src/solutions/common/bizz/facebook/__init__.py [hunk 3]
```diff
@@ -21,9 +21,24 @@
...
+@arguments(domain=unicode)
+def get_facebook_app_info(domain):
+    settings = get_solution_server_settings()
+    apps = settings.facebook_apps
+    mapping = {}
+
+    for host, app_id, app_secret in chunks(apps, 3):
+        mapping[host] = (app_id, app_secret)
+
+    return mapping.get(domain.lower().strip())
```
Comment (bart-at-mobicage) [inline src/solutions/common/bizz/facebook/__init__.py]: In this case creating a dict is overhead
```
settings = get_solution_server_settings()
for host, app_id, app_secret in chunks(apps, 3):
    if domain == host:
        return (app_id, app_secret)
return None
```

File: src/solutions/common/bizz/facebook/__init__.py (hunk 2 above)
Comment (abom) [inline src/solutions/common/bizz/facebook/__init__.py]: It's possible if it has no values to unpack, yes.

**Pull request merged**